block-buffer/src/buffer.rs

```diff
@@ -93,6 +94,65 @@ impl<BlockSize: ArrayLength<u8>> BlockBuffer<BlockSize> {
         self.process_data(data, &mut gen_block, set, |f| f(), unreachable);
     }
 
+    /// Process `data` in blocks and write result to `out_buf`, storing
+    /// leftovers for future use.
+    #[inline]
+    pub fn block_mode_processing<'a>(
+        &mut self,
+        mut data: &[u8],
+        buf: &'a mut [u8],
+        mut process: impl FnMut(&mut [Block<BlockSize>]),
+    ) -> Result<&'a [u8], InvalidLength> {
+        let pos = self.get_pos();
+        let rem = self.remaining();
+        let mut blocks_processed = 0;
+        let (_, mut buf_blocks, _) = to_blocks_mut::<BlockSize, U1>(buf);
```

Comment:
Perhaps this instead?

```suggestion
        let mut buf_blocks = to_blocks_mut::<BlockSize, U1>(buf).1;
```